guillotina/contrib/dbusers/groups.py

```diff
@@ -0,0 +1,20 @@
+from guillotina import configure
+from guillotina.auth.groups import GuillotinaGroup
+from guillotina.interfaces import IGroups
+
+
+@configure.utility(provides=IGroups)
+class GroupsUtility:
```

Comment:
IMHO, from just looking at the implementation, it is a bit hard to understand what the utility is intended for (I mean the one that is already defined in guillotina.auth.groups.py)

What about renaming it to `IGetGroupsUtility` or `GroupsSearch`?

Also maybe typing `principal: Optional[IPrincipal]` could help?

Reply:
I'd say that the interface should be similar to the `auth_user_identifier` so we can get a group doing `async find_group(id)`

Reply:
I added the type annotation